### DISCUSSION TITLE
Combining stabilityDays, prCreation: not-pending and grouping

### DISCUSSION THREAD

**Discussion by kipusoep** [renovatebot/renovate#14786] [Request Help]
I can't seem to get my head around this, how to combine stabilityDays, prCreation: not-pending and grouping together and get the result we'd like to see.

So I've set the stabilityDays to 4 and prCreation to not-pending.
Also added a groupName and groupSlug.

So in my case there are 3 updates available. 2 of them were released 5 days ago. 1 of them is released 3 days ago.
I would like to get a PR of the first 2 (which are stable) and leave out the last package (will be included in the next PR).
I can't seem to get this result, also tried different settings for the internalChecksFilter option.
It seems to be the case that a PR will be created with all 3 of the packages (including the 'unstable' one), or have no PR at all.

How should I configure renovate to get the wanted outcome?

Part of the config:
```json
"packageRules": [{
  "excludePackagePrefixes": [ "CompanyName." ],
  "matchUpdateTypes": [ "minor", "patch", "pin", "digest" ],
  "stabilityDays": 4,
  "prCreation": "not-pending",
  "groupName": "all external non-major dependencies",
  "groupSlug": "all-external-non-major"
}]
```

**Comment by rarkins** [comment]
Look at internalChecksFilter

**Reply by kipusoep** [reply]
Yeah I did; "I can't seem to get this result, also tried different settings for the internalChecksFilter option."
I wasn't able to get the outcome with all three possible options.

(I saw https://github.com/renovatebot/renovate/discussions/14378#discussioncomment-2239031)

**Reply by kipusoep** [reply]
So I would think strict is the best option for this scenario?
> strict: All pending releases will be filtered. PRs will be skipped unless a non-pending version is available

The thing is; it seems the wole grouped PR is skipped when 1 of the updates doesn't meet the stabilityDays requirement.

**Reply by kipusoep** [reply]
So I did another test with strict, and it does update the 2 packages that are 'stable' in the branch, but no PR is created while the status is 'green'.

**Reply by rarkins** [reply]
The PR is only created the next time renovate runs after its green. Always look at your debug logs first

**Reply by kipusoep** [reply]
I did find something interesting. When I remove `"prCreation": "not-pending",` it does create a PR.
> Always look at your debug logs first

I **am** looking at the debug logs, but unable to find any hint regarding the PR creation:

```
 INFO: Branch created (repository=Redacted/Redacted, branch=renovate/all-external-non-major)
       "commitSha": "50e34b5dbc70b967709e72eb4296ec034f9ca5e8"

DEBUG: Updating renovate/stability-days status check state to green (repository=Redacted/Redacted, branch=renovate/all-external-non-major)

DEBUG: setBranchStatus(renovate/all-external-non-major, renovate/stability-days, Updates have met stability days requirement, green, https://docs.renovatebot.com/) (repository=Redacted/Redacted, branch=renovate/all-external-non-major)

DEBUG: Branch status pending (repository=Redacted/Redacted, branch=renovate/all-external-non-major)
       "commitSha": "50e34b5dbc70b967709e72eb4296ec034f9ca5e8"

DEBUG: getBranchPr(renovate/all-internal) (repository=Redacted/Redacted)

DEBUG: getPrList() (repository=Redacted/Redacted)

DEBUG: getBranchPr(renovate/all-external-non-major) (repository=Redacted/Redacted)

DEBUG: getPrList() (repository=Redacted/Redacted)

DEBUG: Closing Dependency Dashboard (repository=Redacted/Redacted)

DEBUG: Removing any stale branches (repository=Redacted/Redacted)
```

**Reply by viceice** [reply]
branch status pending! The logs clearly say you why it's not creating the PR

**Reply by kipusoep** [reply]
Isn't that a bit conflicting with the line before? "Updates have met stability days requirement, green"
Why is the status pending when it's just been set to green?
Maybe I just don't get it at all, but I'm trying hard to do so.

**Reply by rarkins** [reply]
Is that the only test/status check you have?

**Reply by kipusoep** [reply]
Yes currently it is, all tests/status checks are disabled at the moment for renovate testing purposes.

**Reply by rarkins** [reply]
It would be good if you can add a dummy test at least to better simulate a "real" scenario. There are scenarios where we either already do or plan to ignore Renovate's own status checks when deciding if a branch is "green". If the problem still persists, we can keep looking into it.
